src/common/ConfirmationModal.tsx

```diff
@@ -15,6 +15,30 @@ const ConfirmationModal: React.FC<ConfirmationModalProps> = ({ action, handleCli
   const [displayModal, setDisplayModal] = useRecoilState(showConfirmation);
   // @ts-ignore
   const { header, note } = confirmationHeaders[actionCategory][actionName];
+  const getConfirmButtonText = () => {
```

Comment:
It is a simple function that gives us a constant value based on certain conditions and it will not be touched or used that frequently. I think we can move this function to constants -> myGoals.ts and use it from there. The goal is to keep .tsx files clean as much as possible

Reply:
@Tushar-4781 It is not clear to me what change you are requesting. The changes by @alpeshl resolve the issue.  

I do agree that is would be cleaner to have the context of what the button should say injected into the function as an argument - instead of the function having to know about all the different ways in which it is being used 'on the outside'. Is that what you meant? Or is it just about moving the current code as is to a different place?

Reply:
I think what @Tushar-4781 is suggesting is to keep logic separate from UI aspects. getting button copy based on action context is something can be moved outside of UI file.

Reply:
moved function to myGoal.ts

Reply:
moved function to `myGoal.ts`